inferno-vc/src/Inferno/VersionControl/Server.hs

```diff
@@ -125,58 +120,45 @@ vcServer toHandler =
       Map.fromList <$> (forM hs $ \h -> (h,) <$> Ops.fetchVCObject h)
 
 runServer ::
-  forall proxy m a g env config.
+  forall m env config.
   ( HasField "serverHost" config config T.Text T.Text,
     HasField "serverPort" config config Int Int,
-    VCHashUpdate a,
-    VCHashUpdate g,
+    VCHashUpdate (Ops.Author m),
+    VCHashUpdate (Ops.Group m),
     FromJSON config,
-    FromJSON a,
-    FromJSON g,
-    ToJSON a,
-    ToJSON g,
-    Ops.Deserializable m a,
-    Ops.Deserializable m g,
-    Ops.Serializable m a,
-    Ops.Serializable m g,
-    Ops.InfernoVCOperations VCServerError m,
-    Ord g
+    FromJSON (Ops.Author m),
+    FromJSON (Ops.Group m),
+    ToJSON (Ops.Author m),
+    ToJSON (Ops.Group m),
+    Ops.InfernoVCOperations VCServerError m
```

Comment:
we're stuck with this constraint and VCServerError is not extensible by implementors so made this constraint explicit on the typeclass context